### ISSUE TITLE
Overfit example seems to modify the text_and_images in place

### ISSUE THREAD

**Issue by robertmash2** [lucidrains/transfusion-pytorch#24]
@lucidrains, I love what you're doing with this repo.  I found what I think is a bug maybe?  The `text_and_images` object is prepended and appended in place when loss is computed.

In more detail, when I run the overfit example like so:

```
...
text_and_images = [
    [randint(0, 256, (16,)).cuda(), randn(4, 384).cuda(), randint(0, 256, (8,)).cuda(), randn(6, 384).cuda()],
    [randint(0, 256, (16,)).cuda(), randn(7, 384).cuda(), randint(0, 256, (5,)).cuda(), randn(2, 384).cuda(), randint(0, 256, (9,)).cuda()]
]

loss = model(text_and_images)
...
```
`text_and_images` looks like this prior to computing loss:
![image](https://github.com/user-attachments/assets/2dd48a72-71a7-49e3-ba33-f44f9dedea71)
But after executing the line `loss = model(text_and_images)`, it looks like this:
![image](https://github.com/user-attachments/assets/a3e2f326-55e5-4720-9918-7759f6214c5c)
Note the prepended 256 token and the appended 257 tokens.

This does not occur when I use a dataloader, so I suspect it's a copy by reference/value kind of thing.

**Comment by lucidrains** [comment]
@robertmash2 hey Robert, thanks for this, will make the fix this evening

like you noticed, it shouldn't break anything if coming from dataloader

**Comment by lucidrains** [comment]
@robertmash2 should be resolved
